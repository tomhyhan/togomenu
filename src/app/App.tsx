import React from "react";
import logo from "./logo.svg";
import styles from "./App.module.css";
import { Link, Route, Routes } from "react-router-dom";
import Home from "../component/home/home.component";
import Menu from "../component/menu/menu.component";
import Item from "../component/item/item.component";
import { IMenu } from "../data/menu.data";

interface AppProps {
  menu: IMenu;
}

function App({ menu }: AppProps) {
  return (
    <div
      className={styles.app}
      style={{ backgroundImage: 'url("images/homebg.jpg")' }}
    >
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/menu" element={<Menu menu={menu} />} />
        <Route path="/:category" element={<Item menu={menu} />} />
      </Routes>
    </div>
  );
}

export default App;
