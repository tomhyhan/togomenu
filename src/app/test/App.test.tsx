import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../App";
import reactTestRenderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

// test("renders learn react link", () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

describe("App", () => {
  let appComponent: JSX.Element;
  beforeEach(() => {
    appComponent = (
      <MemoryRouter>
        <App
          menu={{
            appetizer: {},
            tempura: {},
            udon: {},
            yaki: {},
            teriyaki: {},
            bento: {},
          }}
        />
      </MemoryRouter>
    );
  });
  describe("snapshot", () => {
    it("renders app", () => {
      const app = reactTestRenderer.create(appComponent);
      expect(app.toJSON()).toMatchSnapshot();
    });
  });
});
