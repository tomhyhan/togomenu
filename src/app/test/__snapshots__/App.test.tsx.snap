// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders app 1`] = `
<div
  className="app"
  style={
    Object {
      "backgroundImage": "url(\\"images/homebg.jpg\\")",
    }
  }
>
  <div
    className="home"
  >
    <img
      alt="togo icon"
      className="icon"
      src="images/togo_AppIcon2.png"
    />
    <div
      className="logoGrp"
    >
      <img
        alt="togo title"
        className="logo"
        src="images/logoTitle.png"
      />
      <div
        className="locationGrp"
      >
        <div
          className="location"
        >
          <strong>
            SFU Burnaby
          </strong>
        </div>
      </div>
    </div>
    <div
      className="menuGrp"
    >
      <a
        href="/menu"
        onClick={[Function]}
      >
        <img
          alt="togo menu"
          className="menu"
          src="images/viewMenu.png"
        />
      </a>
    </div>
  </div>
</div>
`;
